var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -137,6 +137,7 @@ class Axom(CMakePackage, CudaPackage):
     depends_on("python", when="+devtools")
     depends_on("py-sphinx", when="+devtools")
     depends_on("py-shroud", when="+devtools")
+    depends_on("llvm+clang@10.0.0", when="+devtools")
```

Comment:
Should this be `type='build'`? Or does it actually link to the llvm libraries?

Reply:
... what you don't link to `clang-format`?

Reply:
Is `clang-format` a binary or a library? I'm not familiar with LLVM.

Reply:
Sorry my comments are only funny (well honestly maybe they aren't) if you know that clang-format is an executable that formats C++ code.  I'll add a laughing emoji next time

Reply:
Gotcha. So yeah, this should prob be `type='build'` so that `llvm` can be uninstalled after `axom` is installed.

Reply:
It would be used during development (via `make style`), it might be confusing to people that `clang-format` is referenced inside of our host-config that may not be there anymore.

Reply:
Is it only used when you install `axom` or is also used when you run `axom` later?

Reply:
It (and all of `devtools`) is only used when developers of Axom are developing. `+devtools` is not needed/helpful or recommended for users of Axom.

This package serves two purposes: regular spack package that installs axom for others to use and building Axom's third party libraries and dropping a host-config to be used by our developers.  If you like we can speak about this more in depth over a webex but @becker33 is well versed in our workflow.

Reply:
do `type='build'` deps automatically uninstall -- I don't think thats the case? 

For instance, `cmake` doesn't disappear unless you ask for spack to clean things up in some way?

For context: we definitely don't want it to automatically uninstall.  It will be used during development of axom (which spack also helps us set the stage for) 

Assuming there is no auto-uninstall flagging it as `type='build'` should be ok.

For what it is worth -- most of our uses of this will probably be external (using a system install), so most of the time - uninstall isn't really an important/useful consideration.

Reply:
No, it doesn't auto-uninstall. So it sounds like `type='build'` is correct.